core/dbt/task/run.py

```diff
@@ -300,26 +301,53 @@ def _build_run_model_result(self, model, context):
             message=str(result.response),
             adapter_response=adapter_response,
             failures=result.get("failures"),
+            batch_results=None,
         )
 
     def _build_run_microbatch_model_result(
         self, model: ModelNode, batch_run_results: List[RunResult]
     ) -> RunResult:
-        failures = sum([result.failures for result in batch_run_results if result.failures])
+        batch_results = BatchResults()
+        for result in batch_run_results:
+            if result.batch_results is not None:
+                batch_results += result.batch_results
+            else:
+                # TODO: Should we raise an error here?
+                continue
+
+        if len(batch_results.failed) == 0:
+            status = RunStatus.Success
+        elif len(batch_results.successful) == 0:
+            status = RunStatus.Error
+        else:
+            status = RunStatus.PartialSuccess
+
+        if len(batch_results.failed) == 0:
+            msg = f"Batches: {len(batch_results.successful)} successful"
+        else:
+            msg = f"Batches: {len(batch_results.successful)} succeeded, {batch_results.failed} failed"
```

Comment:
let's make these smaller since they appear very visibly in the log lines within the `[...]` section which is meant to be small. 

Perhaps: 
* "SUCESS" if all batches succeed
* "ERROR" if all batches fail
* "PARTIAL SUCCESS (#success/#total_batches)]"